a10_neutron_lbaas/v2/handler_listener.py

```diff
@@ -88,17 +87,19 @@ def _set(self, set_method, c, context, listener):
                 # If the binding is being deleted and the port isn't https
                 # remove the port and re-create it.
                 if binding.status == constants.STATUS_DELETING:
+                    import pdb
+                    pdb.set_trace()
+
                     if (protocol != c.client.slb.virtual_server.vport.HTTPS):
-                        self._delete_listener(c, context, listener)
+                        self._delete_listener(c, context, listener, listener.protocol)
                         set_method = c.client.slb.virtual_server.vport.create
                 elif self._set_a10_https_values(listener, c, cert_data, binding):
-
                     # If the binding hasn't been created and the port isn't https
                     # remove the port and re-create it.
                     if (binding.status == constants.STATUS_CREATING and
-                        protocol != c.client.slb.virtual_server.vport.HTTPS):
+                            protocol != c.client.slb.virtual_server.vport.HTTPS):
 
-                        self._delete_listener(c, context, listener)
+                        self._delete_listener(c, context, listener, protocol)
```

Comment:
Yep, this is deleting the currently selected protocol to make room for an https port.

The protocol is set to https doen on line 105.